### PR TITLE
Fix cart improvements empty state

### PR DESCRIPTION
**Pull request by tauthomas01**
**Why are these changes introduced?**

Fixes https://github.com/Shopify/dawn/issues/314

The goal of this PR is to improve the empty state of the cart.

image

**What approach did you take?**

- Adjust CSS spacing
- Add Liquid text and login link

**Demo links**

- [Store](https://os2-demo.myshopify.com/cart?preview_theme_id=124570173462)
- [Editor](https://os2-demo.myshopify.com/admin/themes/124570173462/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: templates/cart.json [hunk 1]
```diff
@@ -17,10 +17,17 @@
...
+      "settings": {
+        "title": "Shop best-sellers"
+      }
     }
   },
   "order": [
     "cart-items",
-    "cart-footer"
+    "cart-footer",
+    "featured_products"
```
Comment (martinamarien) [inline templates/cart.json:31]: For consistency, let's use `featured-products`

File: sections/main-cart-items.liquid [hunk 2]
```diff
@@ -15,9 +15,16 @@
...
+    <a href="{{ routes.all_products_collection_url }}" class="button">
       {{ 'general.continue_shopping' | t }}
     </a>
+
+    {%- if shop.customer_accounts_enabled -%}
+      <h3 class="cart__login-title">{{ 'sections.cart.login.title' | t }}</h3>
+      <p class="cart__login-paragraph">
+        {{ 'sections.cart.login.paragraph_html' | t: link: routes.account_login_url }}
+      </p>
+    {%- endif -%}
```
Comment (martinamarien) [inline sections/main-cart-items.liquid:27]: Is this only meant to be visible on the empty cart state? 
If it allows you to checkout faster, shouldn't we surface this to customer with items in their cart as well?

**Review by martinamarien — APPROVED** [review]
Looks good on my end Thomas. I'd recommend we request translations now if content is confirmed so that they come in soon 🤞🏻

File: locales/en.default.json [hunk 3]
```diff
@@ -203,7 +203,11 @@
         "total": "Total",
         "quantity": "Quantity"
       },
-      "update": "Update"
+      "update": "Update",
+      "login": {
+        "title": "Have an account?",
+        "paragraph_html": "<a href=\"{{ link }}\" class=\"link underlined-link\">Log in</a> to check out faster."
+      }
```
Comment (martinamarien) [inline locales/en.default.json:210]: Weird. The translation platform says "good to go". But this is new.

File: sections/main-cart-items.liquid (hunk 2 above)
Comment (tyleralsbury) [reply sections/main-cart-items.liquid:27]: Yeah, this is empty state only.  It's an interesting point, though.

File: locales/en.default.json [hunk 4]
```diff
@@ -379,4 +383,4 @@
       "print_gift_card": "Print"
     }
   }
-}
+}
```
Comment (tyleralsbury) [inline locales/en.default.json:386]:
```suggestion
}

```

**Review by tyleralsbury — COMMENTED** [review]
Looks good overall.  One question, can we default the featured collection to the `all_products` collection?  I think that should be possible.  It would allow for merchants to not need to make any changes for it to already look good in the cart.

**Comment by martinamarien** [comment]
> I added a comment regarding that I am unsure if it's possible to display all products. Usually, we need to use an existing collection.

I wonder if we should do this if this is the _only_ place we are doing something like this on the theme.

File: sections/main-cart-items.liquid (hunk 2 above)
Comment (tauthomas01) [reply sections/main-cart-items.liquid:27]: cc. @melissaperreault if you have time to see Martina's question.

File: locales/en.default.json (hunk 3 above)
Comment (tauthomas01) [reply locales/en.default.json:210]: Thanks for the help for troubleshooting. I'm requesting them now 👍

**Comment by tauthomas01** [comment]
> If "all products" isn't possible, maybe we can default it to their first collection, whatever that is? Not sure if that works or not. 

Usually, we use the `frontpage` (homepage) collection, but os2-demo removed it so I am not sure if it's reliable.

File: sections/main-cart-items.liquid [hunk 5]
```diff
@@ -15,9 +15,16 @@
 
   <div class="cart__warnings">
     <h2 class="cart__empty-text">{{ 'sections.cart.empty' | t }}</h2>
```
Comment (svinkle) [inline sections/main-cart-items.liquid:17]: Swap `h2` for `h1`.

File: sections/main-cart-items.liquid [hunk 6]
```diff
@@ -15,9 +15,16 @@
...
-    <a href="{{ routes.all_products_collection_url }}" class="link">
+    <a href="{{ routes.all_products_collection_url }}" class="button">
       {{ 'general.continue_shopping' | t }}
     </a>
+
+    {%- if shop.customer_accounts_enabled -%}
+      <h3 class="cart__login-title">{{ 'sections.cart.login.title' | t }}</h3>
```
Comment (svinkle) [inline sections/main-cart-items.liquid:23]: Swap `h3` for `h2`.

File: sections/main-cart-items.liquid (hunk 2 above)
Comment (melissaperreault) [reply sections/main-cart-items.liquid:27]: This is a good question, I'll investigate with the team and come back! 👍

File: sections/main-cart-items.liquid (hunk 5 above)
Comment (melissaperreault) [reply sections/main-cart-items.liquid:17]: +1

File: locales/en.default.json [hunk 7]
```diff
@@ -190,7 +190,7 @@
       "new_subtotal": "New subtotal",
       "note": "Order special instructions",
       "checkout": "Check out",
-      "empty": "Your cart is empty",
+      "empty": "Your cart is empty.",
```
Comment (melissaperreault) [inline locales/en.default.json:193]:
```suggestion
      "empty": "Your cart is empty",
```
@gretchen-willenberg I know this is a complete sentence but it is used as a heading and I believe we don't use period on these, right?

Comment (gretchen-willenberg) [reply locales/en.default.json:193]: +100 Thank you! Yes. No period. I realized what I had said as I was falling asleep last night and fixed the copy doc this morning.

**Review by martinamarien — APPROVED** [review]
Looks good Thomas. All translations are back too 🙌🏻

**Review by tyleralsbury — APPROVED** [review]
Looks awesome.

File: templates/cart.json [hunk 8]
```diff
@@ -17,10 +17,18 @@
...
     }
   },
   "order": [
     "cart-items",
-    "cart-footer"
+    "cart-footer",
+    "featured-products"
   ]
-}
+}
```
[the remark this answers is not in the document]
Comment (tyleralsbury) [reply templates/cart.json:34]: Feel free to keep this so that the approvals don't get removed.  Just noting it.

File: assets/component-cart.css [hunk 9]
```diff
@@ -22,20 +22,28 @@ cart-items.is-empty .title-wrapper-with-link,
...
+}
+
+.cart__login-paragraph a {
+  font-size: inherit;
+}
+
 @media screen and (min-width: 990px) {
   .cart__warnings {
-    padding: 10rem 0 15rem;
+    padding: 7rem 0 1rem;
```
Comment (melissaperreault) [inline assets/component-cart.css:42]:
```suggestion
    padding: 3rem 0 1rem;
```
I missed this white space adjustment, sorry! This is for smaller screens 🙌

**Review by melissaperreault — APPROVED** [review]

**Review by tyleralsbury — APPROVED** [review]

**Pull request merged**